instrumentation/opentelemetry-api-beta/src/main/java/io/opentelemetry/auto/instrumentation/opentelemetryapi/trace/Bridging.java

```diff
@@ -169,6 +172,12 @@ public void consume(String key, AttributeValue attributeValue) {
         .build();
   }
 
+  public static io.grpc.Context toShaded(Context context) {
+    ContextStore<Context, io.grpc.Context> contextStore =
+        InstrumentationContext.get(Context.class, io.grpc.Context.class);
```

Comment:
unfortunately InstrumentationContext is not documented yet (#588), but this won't work here. InstrumentationContext can only be used inside of advice classes because we change those call sites while injecting the advice, e.g. https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/master/agent-tooling/src/main/java/io/opentelemetry/auto/tooling/context/FieldBackedProvider.java#L82-L86

Reply:
Oh bummer. Any pointer on how to convert this `Context`? :) If it's a hard problem, I can ask about reverting the API change, it seems like a very minor convenience.

Reply:
for this PR i suggest

passing `ContextStore` all the way down to the `Span.Builder` starting here: https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/master/instrumentation/opentelemetry-api-beta/src/main/java/io/opentelemetry/auto/instrumentation/opentelemetryapi/OpenTelemetryApiInstrumentation.java#L68

similar to https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/master/instrumentation/opentelemetry-api-beta/src/main/java/io/opentelemetry/auto/instrumentation/opentelemetryapi/OpenTelemetryApiInstrumentation.java#L90

and let's discuss what to do in the future in #752 👍

Reply:
Thanks, plumbed